[Fix-7236][common] Fix bug of the host cannot be recognized when the endpoint uses the domain name

liqiquan opened the pull request:
…endpoint uses the domain name

<!--Thanks very much for contributing to Apache DolphinScheduler. Please review https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html before opening a pull request.-->

fix bug：#7236

## Purpose of the pull request

<!--(For example: This pull request adds checkstyle plugin).-->

## Brief change log

<!--*(for example:)*
  - *Add maven-checkstyle-plugin to root pom.xml*
-->
## Verify this pull request

<!--*(Please pick either of the following options)*-->

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

<!--*(example:)*
  - *Added dolphinscheduler-dao tests for end-to-end.*
  - *Added CronUtilsTest to verify the change.*
  - *Manually verified the change by testing locally.* -->

zhongjiajie committed:
[deploy] Add tools to distribute package (#7916)

This patch add missing tools diretory in binary
package

close: #7886

Commit:
Improvement: improve Pong.java mistake (#7822)

Ivan0626 committed:
Close the FileOutputStream in FileUtils.java file (#7937)

Close the FileOutputStream in FileUtils.java file

labbomb committed:
[Feature][UI Next]Added the tenant management page (#7945)

* Enabling Route Forwarding
* Added the tenant management page
* tenant-modal

caishunfeng committed:
[E2E] Add queue e2e test (#7941)

calvinjiang committed:
[Fix-7935] Rectify this issue with running test cases in the class of AlertGroupControllerTest. (#7954)

* fix a few issues

calvinjiang committed:
[Fix-7288] Not allow to delete the default alarm group. (#7932)

SbloodyS committed:
[E2E] [Resource] add file manage case (#7906)

zhuangchong committed:
Create or update users to add tenants cannot be empty verification. (#7594)

zhuangchong committed:
[Feature-7804][Task] DingTalk alert plugin adds signature. (#7805)

* DingTalk alarm component adds signature.

* fix code style.

* fix code style.

* add dingtalk document link

* fix DingTalkAlertChannelFactoryTest

* update NAME_DING_TALK_SECRET

songjianet committed:
[Feature][UI Next] Add project list. (#7961)

songjianet committed:
[Fix][UI Next] Fix dark theme logout. (#7965)

SbloodyS committed:
[Fix-7885][Deploy] Output log should under logs/ directory (#7960)

* fix bug_7885

* fix bug_7885

* fix bug_7885

* fix bug_7885

songjianet committed:
[Feature][UI Next] Delete edit profile state. (#7966)

ruanwenjun committed:
Fix global params cannot transport into sub_process (#7969) (#7973)

labbomb committed:
[Feature][UI Next]Improve the tenant Modal (#7972)

* Enabling Route Forwarding

* Added the tenant management page

* tenant-modal

* Improve the tenant Modal

* Enabling Route Forwarding

* Improve the tenant Modal

* fix bug

shangeyao committed:
[Improvement][MasterServer] process complement data optimization #7925 (#7976)

* [Improvement][MasterServer] process complement data optimization #7925

* [Improvement][MasterServer] process complement data optimization #7925

Co-authored-by: shangeyao <sgy960921>

labbomb committed:
[Feature][UI Next]Improve the tenant Modal (#7983)

* Enabling Route Forwarding

* Added the tenant management page

* tenant-modal

* Improve the tenant Modal

* Enabling Route Forwarding

* Improve the tenant Modal

* fix bug

* Improve the tenant Modal

* Improve the tenant Modal

* Improve the tenant Modal

ShuoTiann committed:
[Fix-7984] Add Unit Test for UiPluginController (#7985)

calvinjiang committed:
[Fix-7977] Rectify the wrong prompt in the class of 'TaskGroupServiceImpl'. (#7978)

zhuangchong committed:
Fix query workflow history version information error (#7988)

songjianet committed:
[Feature][UI Next] Add project workflow monitor. (#7991)

devosend committed:
[Feature][UI Next] Add file manage (#7943)

* add file
* fix form style
* fix menu key not init bug
* add license header
* fix review question
* delete jquery
* modify conditions place
* use pagination components instead of pagination attr of tables.
* fixed table column width
* delete card component title required attr
* add table style

calvinjiang committed:
fix the issue with spelling mistakes (#7995)

zhuangchong committed:
fix processe definitionv versions delete failed (#7994)

labbomb committed:
[Feature][UI Next] Improve internationalization of tenant management (#7997)

devosend committed:
[Feature][UI Next] Delete Conditions Component (#7999)

* delete condition component
* rename file manage route

Narcasserun committed:
[fix-7671][plugin] Supports whether SQL is placed in the same session for configuration(fix-7671) (#7675)

-Supports whether SQL is placed in the same session for configuration

labbomb committed:
[Feature][UI Next]Improve tenant manage (#8001)

* Improve tenant manage

* [Feature][UI Next] Delete Conditions Component (#7999)

* delete condition component
* rename file manage route

* [fix-7671][plugin] Supports whether SQL is placed in the same session for configuration(fix-7671) (#7675)

-Supports whether SQL is placed in the same session for configuration

* Add NSpace

* Repair internationalization

* Improve tenant manage

* Add NSpace

* Repair internationalization

* Fix tenant management problems

Co-authored-by: Devosend <devosend@gmail.com>
Co-authored-by: mask <39329477+Narcasserun@users.noreply.github.com>

devosend committed:
[Fix-7717] [UI Next] Prettier config name error (#8012)

* code format

* add package.json change

* code prettier

SbloodyS committed:
[CI][E2E] Optimize CI E2E test performance (#7974)

devosend committed:
[python] Fix global params bug (#7971)

yimaixinchen committed:
Fix ResponseCache Class Name ERROR (#8015)

* fix ResponseCache name error

Commit:
[Bug-7950] Add unit test for AlertPluginInstanceController (#7952)

Commit:
[Enhancement-8006] Improve the performance of SQL query (#8007)

Co-authored-by: kuanghc1 <kuanghc1@lenovo.com>

songjianet committed:
[Feature][UI Next] Add security yarn queue manage. (#8040)

calvinjiang committed:
fix a little issue (#8039)

SbloodyS committed:
[Fix-7513] Fix identical numbers and add positive integer judgment (#8032)

* fix bug_7513

* fix bug_7513

* fix bug_7513

* fix bug_7513

* fix bug_7513

* fix bug_7513

* fix bug_7513

* fix bug_7513

* fix bug_7513

* fix bug_7513

* fix bug_7513

* fix bug_7513

* fix bug_7513

* fix bug_7513

* fix bug_7513

* test

* test

* test

* test

* test

* chinese change

caishunfeng committed:
[conf] Update master config on k8s (#8045)

Co-authored-by: caishunfeng <534328519@qq.com>

songjianet committed:
[Fix][Dag] Timeout alert. (#8049)

Lucaszlei committed:
[DS-6820][feat] Ordinary users can also share resource data source projects with others (#7929)

Admin's entire contents of non-selected users
  Non-admins can only authorize content they create

This closes #6820

zhongjiajie committed:
[e2e] Add new test of security environment (#8048)

* [e2e] Add new test of security environment

ref: #7281

* Fix frontend bug

* Fix order of UT

* Fix delete and edit bottom class

* Try to fix

* Try to fix

songjianet committed:
[Feature][UI Next] Add security environment manage. (#8057)

Amy0104 committed:
[Feature][UI Next] Add data source (#8058)

labbomb committed:
Packaging problem fix (#8059)

labbomb committed:
Fixed error reporting for component JsonHighlight (#8061)

devosend committed:
[Feature-7346] [Python]Add workflow as code task type spark (#7968)

* add spark task

* fix code format

* add parent class for flink and spark

* modify Engine docstring

* modify docstring of Engine

calvinjiang committed:
fix a few issues (#8063)

calvinjiang committed:
[Feature-7785][UI Next] Add a new page for the module of next ui to manage the task group option. (#8060)

* add a few settings

* add a few pages

* modify locals

* modify pages

* add taskGroupOption

* merge from dev

* modify taskGroupOption

* modify task group

* request api

* modify taskGroupOptions

* modify taskGroupOption

* modify task group option

* merge from dev

* prettier the codes

* fix a little issue

* merge from dev

songjianet committed:
[Feature][UI Next] Security alarm group manage (#8069)

* [Feature][UI Next] Add security alarm group manage.

* [Feature][UI Next] Add security alarm group manage license.

labbomb committed:
Fixed Dart-Sass installation error (#8071)

jim-parsons committed:
[Improvement-7907][refactor] Optimization for query task instances list when build dag flow (#7915)

Optimization for query task instances list when build dag flow
This closes #7907

asdf2014 committed:
[Feature][Github] Add stale mechanism to auto mark issues and PRs as stale (#8075)

caishunfeng committed:
[2.0.3][Bug-7986] Disable deletion of running workflow (#8065)

Co-authored-by: caishunfeng <534328519@qq.com>

calvinjiang committed:
[Fix-8056] Rectify this issue about failing to edit worker groups for an environment. (#8072)

devosend committed:
[Feature][UI Next] Add Udf resource manage (#8076)

* add resource manange

* add resource delete

* modify rename

* support sub udf resource mannage

* add Bread

* adjust style

zhongjiajie committed:
Revert "[Feature][Github] Add stale mechanism to auto mark issues and PRs as stale (#8075)" (#8079)

This reverts commit 6b9198acfcf8b61f554295edd0e09d07d674bc8e.

XiaochenNan committed:
add unit test of queryWorkAddressLest (#8054)

Co-authored-by: brian <flyunrain@163.com>

calvinjiang committed:
[Feature-7785][UI Next] Rectify the issue about missing type in the feature of  task group option. (#8083)

* prettier the codes

* fix a little issue

songjianet committed:
[Feature][Workflow Instance] Add Workflow instalce batch delete reload table list. (#8082)

songjianet committed:
[Feature][UI Next] Feat worker group manage (#8087)

* [Feature][UI Next] Add worker group manage.

* [Feature][UI Next] Add worker group manage table tag.

songjianet committed:
[Feature][UI Next] Add monitor router. (#8088)

SbloodyS committed:
[Feature-7993][E2E] Replace HDFS image with MinIO S3 image (#8066)

Commit:
[Bug-8053] Fix CronUtils.getMaxCycle return null (#8086)

seagle-yuan committed:
[bug] Fix datax task type error parse clickhouse reader sql (#8091)

caishunfeng committed:
[deploy] Add datasource driver name into env (#8100)

Co-authored-by: caishunfeng <534328519@qq.com>

songjianet committed:
[Feature][Dag] Number of failed retries and failed retry interval use input component. (#8104)

songjianet committed:
[Feature][UI Next] Add token manage. (#8105)

songjianet committed:
[Feature][UI Next] Add navbar username. (#8108)

songjianet committed:
[Fix][UI Next] Fix the router problem of click to log out. (#8109)

songjianet committed:
[Feature][UI Next] Add graph component. (#8114)

labbomb committed:
Fix menu bar status bug (#8118)

SbloodyS committed:
[E2E][CI] Add DataSource E2E test (#8085)

wangyizhi1 committed:
[Feature][UI Next] Workflow create (#8123)

songjianet committed:
[Feature][UI Next] Add workflow relation layout and format. (#8125)

devosend committed:
[Feature][UI Next] Add udf function manage (#8126)

songjianet committed:
[Feature][UI Next] Organize folders. (#8127)

devosend committed:
[Fix-8098] [UI] Fix resource manage bug (#8107)

wangyizhi1 committed:
[Feature][UI Next] Adjust the projects directory (#8128)

wangyizhi1 committed:
[Feature][UI Next] Modify route registration rules (#8137)

* [Feature][UI Next] Modify route registration rules

* [Feature][UI Next] Delete logs

labbomb committed:
[Feature][UI Next]Improve menu bar function (#8141)

* Improve menu bar function

* Improve menu bar function

SbloodyS committed:
[Fix-8119][API] Update ProcessInstance error and without reason output (#8122)

* fix bug_8119

* fix bug_8119

* fix bug_8119

* fix bug_8119

* fix bug_8119

kezhenxu94 committed:
Fix wrong Python gateway docker image name (#8143)

devosend committed:
[Feature-7348] [Python] Add workflow as code task type mr (#8140)

ljmiaoo committed:
[Feature][UI Next] Security User Manage (#8133)

zhuangchong committed:
[Feature-8138][Task] add at@ msg in the dingtalk task plugin (#8139)

ljmiaoo committed:
[Fix][UI Next] Fix Security User Manage route was outdated (#8153)

songjianet committed:
[Feature][UI Next] Add project workflow relation. (#8155)

* [Feature][UI Next] Add project workflow relation.

* [Feature][UI Next] Add tooltip component.

calvinjiang committed:
[Fix-8158][UI Next] improve the structure of directories (#8159)

chengshiwen committed:
[Improvement][task-plugin] Remove unreasonable log (#8115)

labbomb committed:
Improve the routing (#8176)

caishunfeng committed:
[Bug-7865][MasterServer] retry logic optimization (#8156)

* submit task optimization

* cloneAndReset

* update

* tolerant task restart when init

* fix snoar check

* fix test

* delete unuse file

* taskInstance key

* code style

* skip complete when retry

Co-authored-by: caishunfeng <534328519@qq.com>

qE0a44OX6s committed:
[Feature-7992][Alert] Support PagerDuty Plugin  (#8120)

Co-authored-by: qingwli <qingwli@cisco.com>

zhuangchong committed:
change version to 2.0.4-SNAPSHOT (#8188)

caishunfeng committed:
Fix start standalone fail because db not init yet (#8191)

ruanwenjun committed:
[IMPROVEMENT-8178] Add Netty processor in Spring container (#8179)

* Add Netty processor in Spring container

SbloodyS committed:
[E2E][CI] Add renaming file management E2E tests (#8160)

yc322 committed:
[Feature-5822][api] Add audit log (#6322)

* rebase

* update audit log type name

* fix the quoted property error

* fix bug

* update audit global switch default value

* fix wrong ddl comment

* fix wrong code style

* update audit log search sql

* add licenses

* add unit test

* fix wrong code style

* fix bugs

* fix wrong code style

* add test case

* add test case

* add license

* Add unit test in pom.xml

* resolve code smell problem

* mysql and postgresql ddl update

* update audit log path and audit page query

* update audit configuration

* update audit log schema, add resource type and resource id

* update audit query and list page

* resolve audit log front page bugs and log query error

* update audit configuration

* resolve wrong configuration

* update AuditLogControllerTest

* update application.yaml

* resolve code smell

* update standalone configuration

* resolve bugs

Co-authored-by: yc322 <1393242780@qq.com>

ruanwenjun committed:
[task] Python task support PYTHON_HOME (#8196)

zhongjiajie committed:
[python] Initiate document for pydolphinscheduler (#8005)

ouyangyewei429 committed:
to improvement #8192 (#8193)

Co-authored-by: ouyangyewei <yewei.oyyw@alibaba-inc.com>

kezhenxu94 committed:
[E2E] Clean up the E2E with Java API to avoid platform sensitive (#8197)

labbomb committed:
[Feature][UI Next] Remove sidekey, resolve the sidebar route jump issue (#8200)

caishunfeng committed:
fix host update processor init (#8205)

Co-authored-by: caishunfeng <534328519@qq.com>

Amy0104 committed:
[Feature][UI Next] Add alarm instance manage (#8208)

songjianet committed:
[Fix][UI Next] Fix alert plugin api. (#8211)

SbloodyS committed:
[E2E]Fix file manage e2e error (#8214)

zixi0825 committed:
[Feature][DataQuality] Add Data quality Module #4283 (#6718)

* add data quality module

* add license

* add package configuration in dist pom

* fix license and jar import bug

* replace apache/skywalking-eyes@9bd5feb SHA

* refacotr jbdc-connector and writer

* modify parameter name in HiveConnector

* fix checkstyle error

* fix checkstyle error in dolphinschesuler-dist

* fix checkstyle error in dolphinschesuler-dist

* fix checkstyle error in dolphinschesuler-dist

* fix duplicate code bug

* fix code style bug

* fix code smells

* add dq relevant enums and parameter

* replace apache/skywalking-eyes@9bd5feb SHA

* fix Constants bug

* remove the unused class

* add unit test

* fix code style error

* add unit test

* refactor data quality common entity

* fix code style error

* add unit test

* close e2e test

* fix code smell bug

* modify dataquality enum value to 14 in TaskType

* add data qualtiy task

* update

* add getDatasourceOptions interface

* fix checkstyle

* close e2e test

* add data quality task ui

* update skywalking-eyes SHA

* fix style

* fix eslint error

* fix eslint error

* test e2e

* add unit test and alter dataquality task result

* fix checkstyle

* fix process service test error

* add unit test and fix code smells

* fix checkstyle error

* fix unit test error

* fix checkstyle error

* change execute sql type name

* revert ui pom.xml

* fix data quality task error

* fix checkstyle error

* fix dq task src_connector_type ui select bug

* fix spark rw postgresql bug

* change mysql driver scope

* fix form-create json bug

* fix code smell

* fix DolphinException Bug

* fix ui validate rule and Alert title

* fix target connection param bug

* fix threshold validate change

* add rule input entry index

* change statistic_comparison_check logic

* remove check type change

* add DateExpressionReplaceUtil

* fix null point expetion

* fix null point expetion

* fix test error

* add more sql driver

* fix test error and remove DateExprReplaceUtil

* add get datasource tables and columns

* add get datasource tables and columns

* remove hive-jdbc in pom.xml

* fix code smells

* update sql

* change the pom.xml

* optimize multi_table_accuracy ui

* fix v-show error

* fix code smells

* update sql

* [Feature][DataQuality] Add data quality task ui (#5054)

* add data quality task ui

* update skywalking-eyes SHA

* fix style

* fix eslint error

* fix eslint error

* test e2e

* fix dq task src_connector_type ui select bug

* fix threshold validate change

* remove check type change

* add get datasource tables and columns

* optimize multi_table_accuracy ui

* fix v-show error

* fix code smells

Co-authored-by: sunchaohe <sunzhaohe@linklogis.com>

* [Feature][DataQuality] Add data quality module  (#4830)

* add data quality module

* add license

* add package configuration in dist pom

* fix license and jar import bug

* replace apache/skywalking-eyes@9bd5feb SHA

* refacotr jbdc-connector and writer

* modify parameter name in HiveConnector

* fix checkstyle error

* fix checkstyle error in dolphinschesuler-dist

* fix checkstyle error in dolphinschesuler-dist

* fix checkstyle error in dolphinschesuler-dist

* fix duplicate code bug

* fix code style bug

* fix code smells

* update

* close e2e test

* fix spark rw postgresql bug

* change mysql driver scope

* add more sql driver

* remove hive-jdbc in pom.xml

* change the pom.xml

Co-authored-by: sunchaohe <sunzhaohe@linklogis.com>

* [Feature][DataQuality] Add data quality task backend (#4883)

* add dq relevant enums and parameter

* replace apache/skywalking-eyes@9bd5feb SHA


Co-authored-by: sunchaohe <sunzhaohe@linklogis.com>

* refactor data_quality_module

* add header license

* data quality module refactor

* fix unit test error

* fix checkstyle error

* fix unit test error

* fix checkstyle error

* fix unit test error

* fix code smell

* fix check style

* fix unit test error

* task statistics value add unique code

* fix unit test error

* fix checkstyle error

* fix checkstyle

* fix security hotspot

* fix unit test error

* fix security hotspot

* fix check

* add data quality task error handling

* fix unit test error

* add unit test

* add unit test

* optimize data quality result alert

* fix unit test

* fix sql script error

* fix bug

* update sql script

* fix checkstyle

* add license

* fix checkstyle

* fix checkstyle

* fix unit test

* add jacoco dependencies

* fix unit test

* fix unit test

* add jacoco dependencies

* add unit test

* add unit test

* add license

* fix checkstyle

* fix pom

* fix checkstyle

* fix checkstyle

* merge dev

* fix ui error

* fix pom error

* fix pom error

* fix test error

* fix test error

* mssql-jdbc exclude azure-keyvault

* fix test error

* merge dev and add unit test

* add notes

* rollback the CollectionUtils

* fix

* update sql

* fix

* fix

* fix query rule page error

* change dq.jar path

* fix sql error

* fix ui error

* fix(dq): jar path&task enum description

* add notes on DataQualityApplication

* fix dq result jump error

* fix(ui): page condition

* feat(ui): add show error output path

* change version

* remove all chinese word in sql

* merge

Co-authored-by: sunchaohe <sunzhaohe@linklogis.com>

SbloodyS committed:
[E2E] add udf manage test (#8203)

songjianet committed:
[Feature][UI Next] Change how env is used. (#8221)

caishunfeng committed:
[Improvement][E2E] use testcontainers host (#8224)

songjianet committed:
[Feature][UI Next] Add project task instance. (#8220)

songjianet committed:
[Fix][UI Next] Fix env. (#8229)

zhongjiajie committed:
[python] Clean deps and prepare release (#8210)

* Change package name
* Migrate requirement*.txt to setup.py
* Add extra required for dev
* Add doc RELEASE and DEVELOP
* Correct description

SbloodyS committed:
[E2E] Fix flaky udf manage e2e test (#8232)

* fix_file_manage_e2e_error

* fix error

* fix error

* fix error

brave-lee committed:
[Cherry-pick-dev][API] fix relation api bug (#8207)

* pick 8111

* fix relation delete (#8190)

* pick 8190

SbloodyS committed:
fix_flaky_worker_group_e2e_test (#8237)

labbomb committed:
Fix the routing problem of project management module (#8236)

songjianet committed:
[Feature][UI Next] Add monitor statistics audit log. (#8238)

songjianet committed:
[Fix][UI Next] Fix the problem that the switch language options are not updated. (#8239)

Amy0104 committed:
[Feature][UI Next] Add monaco editor to form (#8241)

devosend committed:
[FIX-7514] press Enter key get "true" not new line (#7596)

Commit:
[Bug-8169] [dolphinscheduler-ui] Fix `After you save the workflow, it… (#8198)

calvinjiang committed:
[Feature][UI Next] Add the page of task group queue. (#8243)

songjianet committed:
[Feature][UI Next] Default open task group manage and project task menu. (#8244)

zhongjiajie committed:
Change bug-report add 2.0.4 and rm 2.0.0 and 1.3.6 (#8234)

qE0a44OX6s committed:
[DS-8215][feat] Support WebexTeams Plugin (#8217)

Co-authored-by: qingwli <qingwli@cisco.com>

labbomb committed:
Fix the routing problem of the user management module (#8251)

zhongjiajie committed:
[python] Fix python api can't connect to remote gateway server (#8248)

For now, python API could only communicate python gateway server
in the same hosts, this patch makes it could work with different hosts,
and export java gateway setting to configure file

Co-authored-by: kezhenxu94 <kezhenxu94@apache.org>
Co-authored-by: ruanwenjun <861923274@qq.com>

SbloodyS committed:
[E2E] add function manage test case (#8250)

QuakeWang committed:
[E2E] Add new test of token management (#8161)

Amy0104 committed:
[Feature][UI Next] Add custom parameters to form (#8271)

calvinjiang committed:
fix this issue (#8272)

calvinjiang committed:
fix this issue (#8273)

devosend committed:
fix udf resource manage router (#8268)

devosend committed:
[Feature][UI Next] Add  Process Definition (#8266)

songjianet committed:
[Feature][UI Next] Add task result. (#8277)

calvinjiang committed:
fix this issue (#8274)

calvinjiang committed:
fix this issue (#8280)

Amy0104 committed:
[Feature][UI Next] Add switch,select and input-number to form (#8287)

SbloodyS committed:
[E2E] Add hive datasource e2e test and fix s3 docker image latest tag (#8276)

songjianet committed:
[Build][UI Next] Update naive-ui. (#8288)

pinkhello committed:
[Feature-8245][Alert] Add Alert Plugin Telegram (#8246)

songjianet committed:
[Feature][UI Next] Add task definition. (#8297)

* [Feature][UI Next] Add task definition.

* [Feature][UI Next] Add task definition.

wangyizhi1 committed:
[Feature][UI Next] Auto layout modal (#8298)

zhuangchong committed:
[Improvement-8284][Alert] Dingtalk alert plugin supports markdown message type (#8285)

* add msgtype in the dingtalk alert plugin

* update markdown msgtype 'at persion'

zhongjiajie committed:
[python][chore] Add pypi related info badge to readme (#8294)

yimaixinchen committed:
change to autowired annotation (#8240)

QuakeWang committed:
[E2E] Add the update test for tenant (#8282)

ljmiaoo committed:
[Fix][UI Next] Fix table of user-manage did not switch language correctly (#8302)

* [Fix][UI Next] Fix Security User Manage route was outdated

* [Fix][UI Next] Fix table of user-manage did not switch language correctly

calvinjiang committed:
fix this issue (#8306)

songjianet committed:
[Fix][UI Next] Fix the problem that the left menu of task group management is not displayed. (#8307)

ruanwenjun committed:
Remove unused class (#8299)

songjianet committed:
[Feature][UI Next] Add token expired jump login page. (#8312)

ouyangyewei429 committed:
[Improvement-8218][dolphinscheduler-api] QueryTaskInstanceListPaging API performance improvement (#8223)

lenboo commented on pom.xml:
```diff
@@ -57,7 +57,7 @@
         <java.version>1.8</java.version>
         <logback.version>1.2.3</logback.version>
-        <hadoop.version>2.7.3</hadoop.version>
+        <hadoop.version>2.8.1</hadoop.version>
```
Is it necessary to upgrade this hadoop version ?
To upgrade the jar package, you need to modify both license and note.

liqiquan replied:
The problem is in the org.apache.hadoop:hadoop-ws jar.
Hadoop has officially solved this problem, but it has only been submitted to versions 2.8.x and 3.0.0-alpha1. This is the  [link](https://issues.apache.org/jira/browse/HADOOP-12963)

In order to solve this problem, our company has upgraded hadoop to 2.8.5。
I will modify both license and note later.

calvinjiang committed:
[Feature][UI Next] Add the page of rule management in the module of data quality. (#8317)

calvinjiang committed:
[Fix][UI Next] Fix a few errors in the task group. (#8322)

wangyizhi1 committed:
[Feature][UI Next] DAG backfill (#8325)

* [Feature][UI Next] Dag backfill

* [Feature][UI Next] Add license header

caishunfeng committed:
[E2E] add switch task case (#8180)

* fix switch bug

* add switch e2e test

* add switch workflow

* pre task option

* selectPreTask id

* update

* switch task e2e

* naming optimization

Co-authored-by: caishunfeng <534328519@qq.com>

calvinjiang committed:
fix this issue (#8332)

ruanwenjun committed:
Optimize some code (#8324)

songjianet committed:
[Docs][UI Next] Add e2e list docs. (#8336)

Amy0104 committed:
[Feature][UI Next] Add SHELL into project(#8339)

wangyizhi1 committed:
[Feature][UI Next] Workflow save modal (#8341)

* [Feature][UI Next] Fix tslint bugs

* [Feature][UI Next] Workflow save modal

* [Feature][UI Next] Format code

yimaixinchen committed:
optimize worker taskProcessor logger. (#8242)

edward-yang committed:
[Bug][API] fix ProcessDefinition update error #8334 (#8338)

* [bug] fix task definition delete

* [bug] fix ProcessDefinition update error #8334

* [bug] fix ProcessDefinition update error #8334

Co-authored-by: edward-yang <yangjianh210@gmail.com>

wangyizhi1 committed:
[Fix-8335][UI] Sqoop node change data source type to hive error (#8348)

calvinjiang committed:
[Feature][UI Next] Add the tooltip to display the whole node's name. (#8353)

wangyizhi1 committed:
[Fix-8172][UI] Fix coordinate calculation error (#8346)

* [Fix-8172][UI] Fix coordinate calculation error

* [Fix-8172][UI] Code format

devosend committed:
[Feature][UI Next] Add Workflow Instance (#8356)

* add workflow list manage

* support batch delete

* add condition search

* add interval to update data

* fix table column I18n error

* add icon for table state column

* del redundant comment

* fix delete data paging jump

Amy0104 committed:
[Fix][UI Next] Fix vue-tsc errors and optimize code. (#8358)

songjianet committed:
[Feature][UI Next] Add e2e to login page. (#8357)

calvinjiang committed:
[Fix-8187][UI] Add the function to the module of resource that you can re-upload the file on the page of the file management. (#8359)

* add reupload file

* develop reupload file

* fix this issue

devosend committed:
[Feature][UI Next] Fix a few bug with workflow manage (#8360)

* fix workflow define international bug

* fix workflow name link font color

* fix operation tooltip display bug

wangyizhi1 committed:
[Feature][UI Next] Create workflow (#8362)

devosend committed:
[Fix][UI Next] Fix a few bug with resource manage (#8363)

* fix resource manage list language switch bug

* fix func manage list language switch bug

labbomb committed:
Adding Rights Management (#8368)

brave-lee committed:
pick 8270/8308 (#8355)

hongshuboy committed:
[Bug-8067][LogServer] fix improper use of StringBuilder (#8365)

fix code format and fix spelling errors

cheneychoi committed:
[Bug] [Parameter Context] The preparameter OUT is null  (#8315)

* fix:param context prop value is null

EricGao888 committed:
[Feature-8222][python] move examples into the scope of source package (#8340)

calvinjiang committed:
[Fix][UI] Rectify this issue with missing the re-uploading file button when the file is under the root directory.

songjianet committed:
[Feature][UI Next] Add e2e to datasource page. (#8380)

qianli2022 committed:
[Feature][api] Task can run on kubernetes - k8s client and namespace #8247 (#8252)

* add db operator for namespace

* add k8s client

* add k8s controller

* add namespace ui

* add licenses

* add kubernetes model known-dependencies

* add new licenses

* fix SonarCloud Code Analysis bug

* remove license for standard protocol

* fix variable MemoryStr to memoryStr

Co-authored-by: qianl4 <qianl4@cicso.com>

SbloodyS committed:
[Fix-8337][Master] Process instance can not be kill when task is failure and can be retry (#8387)

* fix bug_8337

* fix bug_8337

* fix bug_8337

* fix bug_8337

* fix bug_8337

* fix bug_8337

* fix bug_8337

* test

* fix dev_bug_8337

* fix dev_bug_8337

* fix dev_bug_8337

* fix bug_8337

songjianet committed:
[Feature][UI Next] Add e2e to security token manage page. (#8392)

Amy0104 committed:
[Feature][UI Next] Add task creation and task editor. (#8393)

calvinjiang committed:
fix this issue (#8399)

SbloodyS committed:
[Fix-Typo] Fix typo in dev branch. (#8395)

* fix bug_8337

* fix bug_8337

* fix bug_8337

* fix bug_8337

* fix bug_8337

* fix bug_8337

* fix bug_8337

* test

* fix dev_bug_8337

* fix dev_bug_8337

* fix dev_bug_8337

* fix bug_8337

* fix typo

Amy0104 committed:
[Fix][UI Next]: Fix the missing delete button in the custom parameters component. (#8403)

songjianet committed:
[Feature][UI Next] Add e2e to security environment manage page. (#8404)

caishunfeng committed:
fix empty upstreamTaskCodes (#8405)

Co-authored-by: caishunfeng <534328519@qq.com>

devosend committed:
[Feature][UI Next] Add Crontab Component (#8409)

* support crontab component

* fix time-modal form data init bug

songjianet committed:
[Feature][UI Next] Add k8s namespace. (#8411)

ShuiMu-peng committed:
fix sudo.enable=false Is invalid (#8388)

brave-lee committed:
fix relation unbinding bug (#8379)

mchcz committed:
[Bug-8170][UI] SUB_PROCESS Clicking into the child node does not react (#8371)

* Fix-8170 SUB_PROCESS Clicking into the child node does not react

* Fix-8170 SUB_PROCESS Clicking into the child node does not react

devosend committed:
[Fix-8412] [UI] Fix workflow definition timing edit bug.

songjianet committed:
[Feature][UI Next] Add e2e to security yarn queue manage page. (#8420)

caishunfeng commented:
Hi @liqiquan , please resolve the conflicts.

labbomb committed:
[Feature][UI Next] Improve modal component, add link function (#8426)

SbloodyS committed:
[Fix-8296][API] Update process definition error (#8408)

* fix bug_8296

* fix bug_8296

* fix bug_8296

* add postgres sql

* add postgres sql

* add postgres sql

songjianet committed:
[Feature][UI Next] Add e2e to security worker group manage page. (#8434)

SbloodyS committed:
add try catch (#8433)

zhuxt2015 committed:
fix jdbc connect parameters can not input '@' and '.' character (#8293)

* fix jdbc connect parameters can not input '@' and '.' character

This closes #8135

* add AbstractDataSourceProcessor class unit test

* add license header

* remove class comment

* optimize unit tests

* modify the variable name

Co-authored-by: 朱学通 <zhuxuetong@gridsum.com>

wangyizhi1 committed:
[Feature][UI Next] Workflow editing and cells deletion capabilities (#8435)

* [Feature][UI Next] Workflow editing and cells deletion capabilities

* [Feature][UI Next] Code format

Amy0104 committed:
[Feature][UI Next] Add process name and pre tasks into SHELL. (#8437)

calvinjiang committed:
[Feature][UI Next] Add the form of python task to the module of next ui. (#8438)

* add the form of python task

* developed the form of python task

labbomb committed:
[Feature][UI Next]add subprocess task  (#8439)

* Improve modal component, add link function

* add subprocess task

* Resolve the conflict

songjianet committed:
[Feature][UI Next] Add e2e to security tenant manage page. (#8440)

Amy0104 committed:
[Feature][UI Next] Add spark into task. (#8442)

Amy0104 committed:
[Feature][UI Next] Add mr into task. (#8444)

songjianet committed:
[Feature][UI Next] Add e2e to security user manage page. (#8443)

devosend committed:
[Feature][UI Next] Add flink task (#8446)

* Add flink task

* delete redundant file

Amy0104 committed:
[Feature][UI Next] Add http into task. (#8447)

SbloodyS committed:
remove worker group lowercase (#8448)

zixi0825 committed:
fix(dq) change comparison_type from fluctuation to avg (#8382)

liqiquan commented:
> Hi @liqiquan , please resolve the conflicts.

ronyang1985 committed:
[Fix-8389][Common] Fix oshi compatibility issues cause it fail to boot on m1 mac (#8390)

labbomb committed:
Fix field errors in SubProcess (#8461)

labbomb committed:
[Feature][UI Next]Add PIGEON Task (#8462)

* Fix field errors in SubProcess

* Add PIGEON Task

brave-lee committed:
[Bug] [API] fix releaseWorkflowAndSchedule api and relation binding api (#8464)

* fix relation unbinding bug

* fix 8463

CalvinKirs committed:
[Improve]Use hibernate validation framework (#4) (#8467)

* [Improve]Use hibernate validation framework
Improve email verification method
Support domain name verification
* Email is at least a second-level domain name

calvinjiang committed:
[Feature][UI Next] Add the form of sql task to the module of next ui. (#8471)

* add the form of sql task

* add the form of sql task

* add the form of sql task

* add the component of input group

* add the component of input group

* The form of sql task is done

calvinjiang committed:
The form of procedure task is done (#8473)

jim-parsons committed:
[Feature-8386][Server] Custom timezone - Add timezone into user info (#8457)

* [Feature-8386][Server] Custom timezone - Add timezone into user info

  1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Add system default timezone like `GMT+08:00` when create user
  3. Add timeZone property when update user info

This completes #8386

* [Feature-8386][Server] Custom timezone - Add timezone into user info

  1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Add system default timezone like `GMT+08:00` when create user
  3. Add timeZone property when update user info
  4. add ddl sql in update sql script

This completes #8386

* [Feature-8386][Server] Custom timezone - Add timezone into user info

 1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Set timeZone null when create user
  3. Add timeZone property when update user info

This completes #8386

* [Feature-8386][Server] Custom timezone - Add timezone into user info

  1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Set timeZone null when create user
  3. Add timeZone property when update user info

This completes #8386

* [Feature-8386][Server] Custom timezone - Add timezone into user info

  1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Set timeZone null when create user
  3. Add timeZone property when update user info

This completes #8386

* [Feature-8386][Server] Custom timezone - Add timezone into user info

1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Set timeZone null when create user
  3. Add timeZone property when update user info

This completes #8386

* [Feature-8386][Server] Custom timezone - Add timezone into user info

1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Set timeZone null when create user
  3. Add timeZone property when update user info

This completes #8386

CalvinKirs committed:
[Improve]Upgrade Apache Parent version to 25 (#8478)

CallMeKingsley97 committed:
[Improvement][Api] add @Override on the Override method (#8452)

devosend committed:
[Feature][UI Next] Add dag menu (#8481)

* add dag menu

* add dag menu click event

* workflow online edit not allowed

labbomb committed:
[Fix][UI Next] Fix styling issues under Project.

ljmiaoo committed:
[Fix][UI Next] fix user-page lacks authorization button (#8491)

* [Fix][UI Next] Fix Security User Manage route was outdated

* [Fix][UI Next] Fix table of user-manage did not switch language correctly

* [Fix][UI Next] fix user-page lacks authorization button

Amy0104 committed:
[Feature][UI Next] Add SQOOP into task. (#8494)

labbomb committed:
[Fix][UI Next] Fix the corresponding incorrect field.

Ivan0626 committed:
fixbug: varPool value is error (#7724)

devosend committed:
[Feature][UI Next] Add workflow instance startup params

devosend committed:
[Feature][UI Next] Add workflow instance variables params (#8505)

* add variables view on workflow instance

* change parameters header on toolbar

songjianet committed:
[Build][UI Next] Update package. (#8504)

SbloodyS committed:
add index (#8500)

caishunfeng committed:
[Bug-8110][WorkerServer] kill all running task before worker stop (#8490)

* kill all running task before worker stop

* kill local process

* remove kill yarn job

* adjust the orders of close

Co-authored-by: caishunfeng <534328519@qq.com>

calvinjiang committed:
[Feature][UI Next] Add the form of the sea tunnel task to the module of next ui. (#8507)

* add the component of seatunnel

* add the form of the sea tunnel task

* developed the form of the sea tunnel task

calvinjiang committed:
fix this issue (#8509)

devosend committed:
[Feature][UI Next] Add workflow instance dag node echo.

wangbowen1024 committed:
[Fix-8394][Server]fix when the data is supplemented business time still date+1 (#8511)

kezhenxu94 committed:
Add a property to enable/disable metrics module (#8484)

boy-xiaozhang committed:
feat: Unify mysql insert value queto (#8502)

devosend committed:
[Feature][UI Next] Add dag refresh button on workflow instance (#8514)

* modify workflow instance interval refresh

* workflow instance add dag refresh button

zhongjiajie committed:
[python] Add mechanism cli only with version as subcommand (#8516)

* Add basic cli mechanism with Click, for now just including
  one single subcommand `version`
* Add general and easy test class in tests/testing/cli, and
  test to version
* Add sphinx-click to general cli docs basic on click

Yao-MR committed:
[Bug] [Task-plugin]  ShellTask add the parentfolder exist check and create if not exit (#8526)

* check the parent dir exist and create it

* fix lint

Co-authored-by: ywang46 <ywang46@paypal.com>

devosend committed:
[Feature][UI Next] Support readonly model for dag

Lucaszlei committed:
[DS-7896][fix] Fixed when a non-admin user creates a udf function, resources other than .jar will be displayed in the UDF resource (#8458)

This closes #7892

caishunfeng committed:
[Feature][Server] Custom timezone - Add timezone handler unified (#8477)

* date convert of timezone

* remove @JsonFormat

* add unit test

* fix time preview in scheduler

* optimization & add env config

Co-authored-by: caishunfeng <534328519@qq.com>

Commit:
[Fix-7236][common] Fix bug of the host cannot be recognized when the endpoint uses the domain name
